up-l1/zenoh.adoc

```diff
@@ -204,15 +193,14 @@ The section provides the details how it works in different message types.
     * sink: `up://dest-authority/dest-entity/dest-version/dest-resource.instance`
 
 2. `UTransport.send` maps sink (**destination UUri**) to Zenoh key
-    * local (w/o dest-authority): `upl/` + `dest-resource-id + dest-entity-id + dest-version`
-    * remote (with dest-authority): `upr/` + `dest-authority-id/` + `dest-resource-id + dest-entity-id + dest-version`
+    * `up/` + `origin-authority/dest-authority/dest-entity/dest-version/dest-resource`
 
 3. Receiver (dest-authority/dest-entity) registers a listener on **destination UUri**
-    * `up://dest-authority/dest-entity/dest-version/dest-resource.instance`
+    * source: `-`
+    * sink: `up://dest-authority/dest-entity/dest-version/dest-resource.instance`
 
 4. which `UTransport.registerListener` maps to Zenoh key
-    * local (w/o dest-authority): `upl/` + `dest-resource-id + dest-entity-id + dest-version`
-    * remote (with dest-authority): `upr/` + `dest-authority-id/` + `dest-resource-id + dest-entity-id + dest-version`
+    * `up/` + `+++*+++/dest-authority/dest-entity/dest-version/dest-resource`
```

Comment:
I feel like this has been settled at this point, I just don't recall the conversation...

So for Notification messages, the latter parts of the Zenoh key (entity, version, resource) are pulled from the destination, not the source?

Would we want the ability to discriminate at the `registerListener` level as to which source the Notification is coming from? If that were the case, I suppose we'd have to put the latter parts (entity, version, resource) from the source and modify the `registerListener` example to show us using the `source` field.

Reply:
> So for Notification messages, the latter parts of the Zenoh key (entity, version, resource) are pulled from the destination, not the source?

Only Publish messages use source as the latter part of Zenoh key. Notification and Request only care about the destination.

> Would we want the ability to discriminate at the registerListener level as to which source the Notification is coming from? If that were the case, I suppose we'd have to put the latter parts (entity, version, resource) from the source and modify the registerListener example to show us using the source field.

I don't quite get your point. Maybe retrieving the source information from uAttributes is enough?